### PR TITLE
findme feature and updated map_center settings

### PR DESCRIPTION
**Pull request by adisa39**
updated map center logics to save directly into UserSettings and Seller tables respectively.
added getUserLocation API endpoint to get authenticated user location from the backend.
attempted to set map center based on user findme preference.

### CODE REVIEW

File: src/models/UserSettings.ts [hunk 1]
```diff
@@ -25,11 +25,12 @@ const userSettingsSchema = new Schema<IUserSettings>(
     image: {
       type: String,
       required: false,
+      default: ''
     },
     findme: {
       type: String,
       required: true,
-      default: 'deviceGPS'
+      default: 'auto'
```
Comment (swoocn) [inline src/models/UserSettings.ts:33]: 👍

**Review by swoocn — APPROVED** [review]
Great job, @adisa39 & @DarinHajou!
It seems you both had a similar vision for the changes regarding the map center, with the other @ [https://github.com/map-of-pi/map-of-pi-backend-react/pull/160](https://github.com/map-of-pi/map-of-pi-backend-react/pull/160).  I decided to proceed reviewing/ approving this PR because it provided a more comprehensive solution while incorporating elements from the other PR.

Additionally, I made a few adjustments to maintain code cleanliness and consistency.  However, it might be worth revisiting potential refactoring in the future.  Thanks again! 😎👍

**Comment by adisa39** [comment]
> Great job, @adisa39 & @DarinHajou! It seems you both had a similar vision for the changes regarding the map center, with the other @ #160. I decided to proceed reviewing/ approving this PR because it provided a more comprehensive solution while incorporating elements from the other PR.
> 
> Additionally, I made a few adjustments to maintain code cleanliness and consistency. However, it might be worth revisiting potential refactoring in the future. Thanks again! 😎👍


alrig

**Comment by DarinHajou** [comment]
Ok, nice! Does everything work as expected? Does the mods stil get the error?

**Comment by swoocn** [comment]
> Ok, nice! Does everything work as expected? Does the mods stil get the error?

Still need to review and deploy the frontend portion. I'll be holding off on deploying this backend change until the frontend is ready, so we can deploy both together and maintain app stability.

**Pull request merged**